teamcity: failed tests on master: acceptance/TestDockerRuby/runMode=docker#01, acceptance/TestDockerRuby/runMode=docker, acceptance/TestDockerRuby

The following tests appear to have failed:

[#396562](https://teamcity.cockroachdb.com/viewLog.html?buildId=396562):

```
--- FAIL: acceptance/TestDockerRuby/runMode=docker#01 (0.000s)
Test ended in panic.

------- Stdout: -------
E171102 17:35:54.385237 1 cli/error.go:74  unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

rpc error: code = Unavailable desc = transport is closing
Error: unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

rpc error: code = Unavailable desc = transport is closing
Failed running "init"
non-zero exit code: 1
panic: non-zero exit code: 1 [recovered]
	panic: non-zero exit code: 1 [recovered]
	panic: non-zero exit code: 1

goroutine 11231 [running]:
testing.tRunner.func1(0xc420829d10)
	/usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x1a7f460, 0xc4203d5ce0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).stopOnPanic(0xc420f05b00, 0x2a2afa0, 0xc4200180c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:267 +0xb6
panic(0x1a7f460, 0xc4203d5ce0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.maybePanic(0x2a13a60, 0xc4203d5ce0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/docker.go:183 +0x4a
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).RunInitCommand(0xc420f05b00, 0x2a2afa0, 0xc4200180c0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:555 +0x283
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).Start(0xc420f05b00, 0x2a2afa0, 0xc4200180c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:678 +0x4d5
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster(0x2a2afa0, 0xc4200180c0, 0xc420829d10, 0x1c33a0e, 0x4, 0xc421964960, 0x1, 0x1, 0x12a05f200, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:182 +0xa8b
github.com/cockroachdb/cockroach/pkg/acceptance.testDocker.func1(0xc420829d10)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:78 +0x2ca
testing.tRunner(0xc420829d10, 0xc4219f2640)
	/usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x2de
--- FAIL: acceptance/TestDockerRuby/runMode=docker (0.000s)
Test ended in panic.

------- Stdout: -------
Cluster successfully initialized
--- FAIL: acceptance/TestDockerRuby (0.000s)
Test ended in panic.


```
Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/cluster/dockercluster.go
+++ b/pkg/acceptance/cluster/dockercluster.go
@@ -542,6 +542,9 @@ func (l *DockerCluster) startNode(ctx context.Context, node *testNode) {
 // automatically, but exposed for tests that use INIT_NONE. nodeIdx
 // may designate any node in the cluster as the target of the command.
 func (l *DockerCluster) RunInitCommand(ctx context.Context, nodeIdx int) {
+	tCtx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
 	containerConfig := container.Config{
 		Image:      *cockroachImage,
 		Entrypoint: cockroachEntrypoint(),
@@ -552,8 +555,25 @@ func (l *DockerCluster) RunInitCommand(ctx context.Context, nodeIdx int) {
 			"--logtostderr",
 		},
 	}
-	maybePanic(l.OneShot(ctx, defaultImage, types.ImagePullOptions{},
-		containerConfig, container.HostConfig{}, "init-command"))
+
+	log.Infof(ctx, "trying to initialize via %v", containerConfig.Cmd)
+	// This is called early in the bootstrap sequence, and the node may not have
+	// opened its ports yet. Retry appropriately.
+	for {
+		err := l.OneShot(
+			tCtx, defaultImage, types.ImagePullOptions{}, containerConfig, container.HostConfig{}, "init-command",
+		)
+		switch err {
+		case context.Canceled:
+			maybePanic(errors.Wrap(err, "while trying to run init"))
+		case nil:
+			log.Info(ctx, "cluster successfully initialized")
+			return
+		default:
+			// Keep going as long as the context allows.
+			log.Infof(tCtx, "retrying after: %s", err)
+		}
+	}
 }
 
 // returns false is the event

